File: play-2.4.x/K-Modules/kmodules/K-Base/app/models/K/EbeanConfig/MysqlEncryptKeyManager.java
package models.K.EbeanConfig;

import K.Common.Helper;
import com.avaje.ebean.config.EncryptKey;
import com.avaje.ebean.config.EncryptKeyManager;
import play.Logger;

public class MysqlEncryptKeyManager implements EncryptKeyManager {

    public MysqlEncryptKeyManager() {
        Logger.debug("Ebean Server load MysqlEncryptKeyManager");
    }

    @Override
    public EncryptKey getEncryptKey(String tableName, String columnName) {

        return new MysqlEncrptKey(tableName, columnName);
    }

    @Override
    public void initialise() {

    }

    class MysqlEncrptKey implements EncryptKey {

        private String tableName;
        private String columnName;

        public MysqlEncrptKey(String tableName, String columnName) {
            this.tableName = tableName;
            this.columnName = columnName;
        }

        @Override
        public String getStringValue() {
            String sv = Helper.Md5OfString(String.format("%s@%s", tableName, columnName));
            return sv;
        }

    }
}

File: play-kotlin-2.5.x/PlayKotlin/kmodules/K-Base/app/models/K/EbeanConfig/MysqlEncryptKeyManager.kt
package models.K.EbeanConfig

import K.Common.Helper
import com.avaje.ebean.config.EncryptKey
import com.avaje.ebean.config.EncryptKeyManager

class MysqlEncryptKeyManager : EncryptKeyManager {

    init {
        Helper.DLog("Ebean Server load MysqlEncryptKeyManager")
    }

    override fun getEncryptKey(tableName: String, columnName: String): EncryptKey {

        return MysqlEncrptKey(tableName, columnName)
    }

    override fun initialise() {

    }

    internal inner class MysqlEncrptKey(private val tableName: String, private val columnName: String) : EncryptKey {

        override fun getStringValue(): String {
            val sv = Helper.Md5OfString(String.format("%s@%s", tableName, columnName))
            return sv
        }

    }
}
